### DISCUSSION TITLE
Minio plugin: You must specify at least one part error

### DISCUSSION THREAD

**Discussion by XMethues** [nocodb/nocodb#8425] [Q&A]
Install both nocodb and minio in docker. Endpoint and ports are local ip ports.
```
[Nest] 11  - 05/08/2024, 11:40:58 AM   ERROR [MissingDBError] S3Error: You must specify at least one part
[Nest] 11  - 05/08/2024, 11:40:58 AM   ERROR [GlobalExceptionFilter] You must specify at least one part
S3Error: You must specify at least one part
    at parseError (/usr/src/app/node_modules/minio/dist/main/internal/xml-parser.js:47:13)
    at Object.parseResponseError (/usr/src/app/node_modules/minio/dist/main/internal/xml-parser.js:94:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Client.makeRequestStreamAsync (/usr/src/app/node_modules/minio/dist/main/internal/client.js:491:19)
    at async Client.completeMultipartUpload (/usr/src/app/node_modules/minio/dist/main/internal/client.js:1162:17)
    at async /usr/src/app/node_modules/minio/dist/main/internal/client.js:1458:14
    at async Promise.all (index 1)
    at async Client.uploadStream (/usr/src/app/node_modules/minio/dist/main/internal/client.js:1412:20)
```

**Comment by macdakilla** [comment]
I also have this error

**Reply by o1lab** [reply]
Would love a PR for this please.

**Reply by DarkPhoenix2704** [reply]
Can you verify if the following build fixes the issue and is compatible with the existing attachments? 


```bash
docker run -d -p 8888:8080 nocodb/nocodb-timely:0.251.3-pr-9104-20240730-0336
```

**Comment by st34k** [comment]
I'm facing the same issue exactly. I was unable to track down the source. Downgraded minio to v7.1.3 for now.

**Reply by celantunes** [reply]
Após o downgrade vc conseguiu conectar o minio no nocodb ?

**Comment by pranavxc** [comment]
I tried Minio locally(latest docker image) and seems fine. @Tatametheus @macdakilla - Please share following informations,

1. NocoDB version you are using.
2. Minio version
3. Are you using file upload or url based upload, if possible share the api call path from networks tab.

**Reply by mattalxndr** [reply]
@pranavxc 

1. [docker.io/nocodb/nocodb:0.251.3@sha256:c40e3fabcd010f1c7658bbaf8d96d00dd533a2aaeeedd597b96e9379298dd2ee](https://hub.docker.com/layers/nocodb/nocodb/0.251.3/images/sha256-cc682a2cb3650815e05f1c23ee1df5bbfa127b2f046ac67f6f077dd6f411225b?context=explore)
2. [quay.io/minio/minio:RELEASE.2024-07-26T20-48-21Z@sha256:9ef3d0514c0636e86e3db6b2a21c21d4035c17a47ea5464658f2f5d24afd15c6](https://quay.io/repository/minio/minio?tab=tags&tag=RELEASE.2024-07-26T20-48-21Z)
3. This error happens when I click "Test Connection" on the Minio connection/auth dialog on the App Store view. No uploading happening yet.

```
nocodb          | S3Error: You must specify at least one part
nocodb          |     at parseError (/usr/src/app/node_modules/minio/dist/main/internal/xml-parser.js:47:13)
nocodb          |     at Object.parseResponseError (/usr/src/app/node_modules/minio/dist/main/internal/xml-parser.js:94:11)
nocodb          |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
nocodb          |     at async Client.makeRequestStreamAsync (/usr/src/app/node_modules/minio/dist/main/internal/client.js:491:19)
nocodb          |     at async Client.completeMultipartUpload (/usr/src/app/node_modules/minio/dist/main/internal/client.js:1162:17)
nocodb          |     at async /usr/src/app/node_modules/minio/dist/main/internal/client.js:1458:14
nocodb          |     at async Promise.all (index 1)
nocodb          |     at async Client.uploadStream (/usr/src/app/node_modules/minio/dist/main/internal/client.js:1412:20)
```

**Reply by mattalxndr** [reply]
> if possible share the api call path from networks tab.

```
curl 'https://nocodb.***.me/api/v1/db/meta/plugins/test' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'accept-language: en-US,en;q=0.9' \
  -H 'cache-control: no-cache' \
  -H 'content-type: application/json' \
  -H 'cookie: *************' \
  -H 'dnt: 1' \
  -H 'origin: https://nocodb.***.me' \
  -H 'pragma: no-cache' \
  -H 'priority: u=1, i' \
  -H 'referer: https://nocodb.***.me/dashboard/' \
  -H 'sec-ch-ua: "Not/A)Brand";v="8", "Chromium";v="126"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-origin' \
  -H 'user-agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36' \
  -H 'xc-auth: *********' \
  -H 'xc-gui: true' \
  --data-raw '{"input":"{\"access_key\":\"****************8\",\"access_secret\":\"*********************\",\"bucket\":\"****-nocodb-storage\",\"endPoint\":\"minio.***.me\",\"port\":\"443\",\"useSSL\":true}","title":"Minio","category":"Storage"}'
```

**Reply by mattalxndr** [reply]
![Screenshot_2024-07-29_11-28](https://github.com/user-attachments/assets/1f97462e-4984-4670-86f9-20848fec10f7)

**Comment by bachkukkik** [comment]
First of all, I am very new to NocoDB. Just try it out no more than a week.

I'm using latest version both NocoDB and minio. I also faced the same issue -- `S3Error: You must specify at least one part` when pressing "Test Connection" in minio installer

I pressed save anyway and here are what I found.

- the file column works fine regardless of the display on NocoDB (I have verified in minio console itself).
- However; if you upload files to private collection in Minio, you won't be able to access that file through NocoDB either. You will need S3 signature to access the file
- But if you upload files to public collection in Minio, you will be able to see thumbnail as well as the file in NocoDB. Also you will be able to use download button.
- It is best to set Minio API server to be accessible publicly so that you can download from anywhere

below are the section of docker-compose.yml regarding minio + create private/public bucket and screenshots of uploaded files

```
  storage:
    image: minio/minio:latest
    command: server /data --console-address ":9001"
    environment:
      - MINIO_ACCESS_KEY=${MINIO_ACCESS_KEY}
      - MINIO_SECRET_KEY=${MINIO_SECRET_KEY}
    healthcheck:
      test: timeout 5 bash -c "</dev/tcp/localhost/9000"
      interval: 5s
      timeout: 5s
      retries: 10
    restart: always
    volumes:
      - storage:/data:rw
    networks:
      default:

  createbuckets:
    image: minio/mc
    entrypoint: >
      /bin/sh -c "
      /usr/bin/mc alias set storage http://storage:9000 ${MINIO_ACCESS_KEY} ${MINIO_SECRET_KEY};
      /usr/bin/mc mb storage/public;
      /usr/bin/mc anonymous set public storage/public;
      /usr/bin/mc mb storage/private;
      exit 0;
      "
    depends_on:
      storage:
        condition: service_healthy
    networks:
      - default
```

* prove if NocoDB can upload files to private bucket

<img width="890" alt="Screenshot 2567-07-29 at 19 20 07" src="https://github.com/user-attachments/assets/8feaa7d3-a17f-4f6d-b598-1d3d8c0fdc39">

* Check thumbnail / file display (in public bucket)
<img width="725" alt="Screenshot 2567-07-29 at 19 22 53" src="https://github.com/user-attachments/assets/3163033b-5c23-497e-bca3-b38d03d3f16e">

* This is how an entry of file type should look like in Postgresql DB
<img width="647" alt="Screenshot 2567-07-29 at 21 08 33" src="https://github.com/user-attachments/assets/c8d70857-f022-4eb7-9492-29c3ffd1e88e">

Thanks for all developers who contribute to this repo. I can feel its potential during my short test-flight

**Reply by o1lab** [reply]
cc : @DarkPhoenix2704

**Reply by DarkPhoenix2704** [reply]
Hey @bachkukkik . Can you verify if the following build fixes the issue and is compatible with the existing attachments? 


```bash
docker run -d -p 8888:8080 nocodb/nocodb-timely:0.251.3-pr-9104-20240730-0336
```
